shared/actions/chat/shared.js

```diff
@@ -170,30 +170,40 @@ function _filterTimestampableMessage (message: Constants.Message): ?Timestampabl
     return message
   }
 
-  if (message === null || message.type === 'Timestamp' || ['Timestamp', 'Deleted', 'Unhandled', 'InvisibleError', 'Edit'].includes(message.type)) {
-    return null
-  }
-
-  if (!message.timestamp) {
-    return null
-  }
+  if (!_isTimestampableMessage(message)) return null
 
   // $FlowIssue with casting todo(mm) can we fix this?
   return message
 }
 
-function maybeAddTimestamp (_message: Constants.Message, _prevMessage: Constants.Message): Constants.MaybeTimestamp {
-  const prevMessage = _filterTimestampableMessage(_prevMessage)
+function _isTimestampableMessage (message: Constants.Message): boolean {
+  if (message && message.timestamp && !['Timestamp', 'Deleted', 'Unhandled', 'InvisibleError', 'Edit'].includes(message.type)) {
+    return true
+  }
+  return false
+}
+
+function _previousTimestampableMessage (messages: Array<Constants.Message>, prevIndex: number): ?Constants.Message {
+  for (var i = prevIndex; i >= 0; i--) {
```

Comment:
findLast doesn't quite work here since I need to start from prevIndex.
findLast would return the last element always.

```
// Always returns same thing
return messages.findLast(m => _isTimestampableMessage(m))
```